### PR TITLE
Refactor maternity-paternity-calculator to remove use of next_node_calculation blocks

### PR DESCRIPTION
**Pull request by floehopper**
## Description

Supersedes #2636.

Trello card: https://trello.com/c/g0Z5s0XE

Similar to #2632, this moves the flow _towards_ the style described in the [refactoring documentation][1]. However, unlike previous similar refactoring PRs, I've focussed exclusively on removing the use of `next_node_calculation` blocks from the flow definition. This is so I can get to the point where I can remove `next_node_calculation` from the DSL as soon as possible. It would be a _lot_ more work to fully refactor this flow to the new-style.

## External changes

None. This is just an internal refactoring.

[1]: https://github.com/alphagov/smart-answers/blob/master/doc/refactoring.md

### CODE REVIEW

**Comment by floehopper** [comment]
This PR is ready for code review.

File: test/unit/calculators/maternity_paternity_calculator_test.rb [hunk 1]
```diff
@@ -483,7 +498,9 @@ class MaternityPaternityCalculatorTest < ActiveSupport::TestCase
         should "calculate pay due for each pay date on a weekly cycle" do
           @calculator.pay_date = Date.parse('28 December 2012') # Friday before maternity leave/pay starts.
           @calculator.pay_method = 'weekly'
-          @calculator.calculate_average_weekly_pay('weekly', 2000)
+          @calculator.pay_pattern = 'weekly'
+          @calculator.earnings_for_pay_period = 2000
+          @calculator.average_weekly_earnings
```
Comment (chrisroos) [inline test/unit/calculators/maternity_paternity_calculator_test.rb]: Is this line (and a few others like it in this file) necessary now that `#average_weekly_earnings` doesn't change the state of the `@calculator`? They were changed in e6601288ac00cd7424c57acc8cb33be44706b803 ("Convert average_weekly_earnings into fully-fledged calculator method").

**Comment by chrisroos** [comment]
In the commit note for 7b9b8c131ce4dd3bc2e93d607e2706f5b94cc7b5 ("Store earnings_for_pay_period as attribute on calculator"), should "I will need both the remaining argument" be "I will need the remaining argument"?

**Comment by chrisroos** [comment]
I've left two comments about minor things but overall this looks good to me, @floehopper.

File: test/unit/calculators/maternity_paternity_calculator_test.rb (hunk 1 above)
Comment (floehopper) [inline test/unit/calculators/maternity_paternity_calculator_test.rb]: No. You're right. Well spotted. I've addressed this in a fix-up commit: 16d7cdcbdf812a1a3a020d33100754c25ff74762.

**Comment by floehopper** [comment]
> In the commit note for 7b9b8c131ce4dd3bc2e93d607e2706f5b94cc7b5 ("Store earnings_for_pay_period as attribute on calculator"), should "I will need both the remaining argument" be "I will need the remaining argument"?

Yes. Thanks. I've added an empty squash commit: 010c741b91f6f4f179bbccf70d313f69d9ee1084 to remind me to fix this when I rebase.

**Comment by floehopper** [comment]
I'm going to rebase this interactively to incorporate the fix-up & squash commits, then I'll rebase against master and force-push in preparation for merging.

**Pull request merged**